File: src/CoralTime.DAL/Models/LogChanges/MemberAction.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using CoralTime.Common.Constants;

namespace CoralTime.DAL.Models.LogChanges
{
    public class MemberAction
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        
        public DateTime Date { get; set; } = DateTime.Now;
        
        public string ChangedObject { get; set; }
        
        public string Entity  { get; set; }
        
        public string ChangedFields { get; set; }
        
        public string Action { get; set; }
        
        public string EntityId { get; set; }
        
        public int MemberId { get; set; }
        
        [ForeignKey("MemberId")]
        public Member.Member Member { get; set; }
    }
}

File: src/CoralTime/ClientApp/src/app/models/member-action.ts
export class MemberAction {
    id: number;
    date: Date;
    changedObject: string;
    entity: string;
    changedFields: ChangedField[];
    action: string;
    entityId: string;
    memberId: number;
    memberFullName: string;

    constructor(data = null) {
        if (data) {
            this.id = data.id;
            this.date = new Date(data.date);
            this.changedObject = data.changedObject
                .replace( /({"|"}|})/g,'')
                .replace(/(","|,")/g, ',  ')
                .replace(/(:"|":"|":)/g, ':');
            this.entity = data.entity
            this.changedFields = JSON.parse(data.changedFields);
            this.action = data.action;
            this.entityId = data.entityId;
            this.memberId = data.memberId;
            this.memberFullName = data.memberFullName;
        }
    }
}

export interface ChangedField {
    field: string;
    oldValue: string;
    newValue: string;
}
